gluten-core/src/main/scala/org/apache/gluten/backendsapi/SparkPlanExecApi.scala

```diff
@@ -672,9 +672,11 @@ trait SparkPlanExecApi {
     def getPushedFilter(dataFilters: Seq[Expression]): Seq[Expression] = {
       val pushedFilters =
         dataFilters ++ FilterHandler.getRemainingFilters(dataFilters, extraFilters)
-      pushedFilters.filterNot(_.references.exists {
-        attr => SparkShimLoader.getSparkShims.isRowIndexMetadataColumn(attr.name)
-      })
+      pushedFilters
+        .filterNot(_.references.exists {
+          attr => SparkShimLoader.getSparkShims.isRowIndexMetadataColumn(attr.name)
+        })
+        .filter(_.deterministic)
```

Comment:
Vanilla Spark would not push down non-deterministic filer exprs. But as Gluten have already supported it, can we add a new config to control if push down non-deterministic filer exprs ?